src/types/interfaces.ts

```diff
@@ -942,16 +952,32 @@ export interface SdkMediaEvents {
    *  depending on the outcome of the `getUserMedia()` request
    */
   permissions: SdkMediaStateWithType;
+
+  /**
+   * Event when the SDK makes a request to `navigator.mediaDevices.getUserMedia()`
+   *  (gUM). Many browsers only allow requests to `gUM()` when the window is in
+   *  focus. Use this event to monitor when media is requested to ensure that the
+   *  window is in focus so the media request will complete.
+   *
+   * Note: if requesting both media types at the same time (`audio & video`), the
+   *  SDK will make _two_ separate `gUM()` requests – one for each media type
+   *  (see notes on `sdk.media.startMedia()`). This event will emit for _each_
+   *  `gUM()` request. For this reason, it is recommended to utilize the
+   *  `uuid` field of the `interface IMediaRequestOptions` to help track which
+   *  `gUM()` requests were made by your application (see docs for `IMediaRequestOptions#uuid`).
+   */
+  gumRequest: ISdkGumRequest;
 }
 
-export type SdkMediaEventTypes = keyof Omit<SdkMediaEvents, 'audioTrackVolume'>;
+export type SdkMediaEventTypes = keyof Omit<SdkMediaEvents, 'audioTrackVolume' | 'gumRequest'>;
```

Comment:
why is gumRequest omitted?

Reply:
This type is only for the `sdk.media.on('state', event)` event. Which will emit for `'permissions' | 'devices' | 'state'`. This was just a quick way to make sure that these types overlapped with the actual `interface SdkMediaEvents` keys.

Reply:
ahh i see